pennylane/ops/qubit/parametric_ops.py

```diff
@@ -97,11 +103,9 @@ def compute_matrix(theta):  # pylint: disable=arguments-differ
             c = qml.math.cast_like(c, 1j)
             s = qml.math.cast_like(s, 1j)
 
+        c = (1 + 0j) * c
```

Comment:
How come this is needed?

Reply:
Will have to check which interface brought this up...

Reply:
So the previous solution to avoid `torch`'s warning about casting complex quantities to reals was avoided by stacking purely real and purely complex quantities and adding the resulting arrays.
The above also avoids this warning, but only requires a 0D (unbroadcasted)/1D(broadcasted) manipulation.
As the broadcasting dimension increases the "stacking effort" we have to pay, the above seems cheaper to me.
What do you think about this? :thinking:

Reply:
nice solution!